diff --git a/src/arch/zynqmp.c b/src/arch/zynqmp.c
index c3a28f0..a80c065 100644
--- a/src/arch/zynqmp.c
+++ b/src/arch/zynqmp.c
@@ -66,6 +66,11 @@ int zynqmp_bootrom_init_header(bootrom_hdr_t *hdr, bootrom_offs_t *offs) {
     hdr->reg_init_zynqmp[(2 * i) + 1] = 0x0;
   }
 
+  /* Update lpd_apu register for cache coherency of AXI HPC */
+  /* See https://xilinx-wiki.atlassian.net/wiki/spaces/A/pages/18842098/Zynq+UltraScale+MPSoC+Cache+Coherency */
+  hdr->reg_init_zynqmp[0] = 0xFF41A040;
+  hdr->reg_init_zynqmp[1] = 0x03;
+
   /* Fill padding */
   memset(hdr->padding, 0xFFFFFFFF, sizeof(hdr->padding));
 
